content/2018-12-18_dvanacty-sraz-matice.md

```diff
@@ -0,0 +1,85 @@
+Title: Dvanáctý sraz - matice
+Date: 2018-12-18 18:00:00
+Modified: 2018-12-18 18:00:00
+Author: Iveta Česalová
+
+<div class="text-justify">Po počátečním zmatku, kdy si Petr zapomněl svůj počítač, jsme poslední sraz roku zahájily opět rekapitulací, co kdo dělal. Doposud byla hra rozdělena na čtyři moduly – backend, frontend, game a testy. Nyní je v backendu pouze třída Robot, State a funkce pro načtení souřadnic, cesty k robotům, počtu políček a počáteční stav hry.</div>
+
+<div class="text-justify">V repozitáři přibylo několik souborů. Loading.py obsahuje funkce, které kompletně načtou mapu se všemi potřebnými komponenty a vlastnostmi a soubor Util.py zase třídy Tile a Direction. Soubor Validator.py kontroluje, zda je mapa správně načtena.</div>
+
+
+####Kouzelné podtržítko####
+<div class="text-justify">Pokud chceme, aby ve zbytku kódu nepoužíval konkrétní proměnnou, dáme před ni podtržítko, jako např u ` _board`. Značí věci, které jsou pro nějakou třídu interní. Podtržítkem říkám – vyhrazuji si právo cokoli změnit.</div>
```

Comment:
aby ve zbytku kódu kdo nepoužíval konkrétní proměnnou?

Reply:
tohle nevím, prosím o pomoc, když mrkneš na záznam tak nevím jak to vysvětlit... a nebo celé smazat :-D @anezkamll

Reply:
Můj komentář směřoval halvně k tomu, že mi v té větě chybí nějaký podmět :) 
Něco jako: "Pokud nechceme, aby někdo ve zbytku kódu používal konkrétní proměnnou, dáme před ni podtržítko..." nebo "Pokud chceme, aby ve zbytku kódu nikdo nepoužíval konkrétní proměnnou, dáme před ni podtržítko..."
Jo a napadá mě k tomu ještě, že v našem případě board není proměnná, ale slovník. 
Možná bych k tomu,  jak píšeš, že jde o věci interní pro nějakou třídu dodala to, co tam zmiňuje Petr - ten, kdo nemá dobrý důvod pracovat s vnitřnostmi té třídy, měl by použít něco jiného.

Reply:
@encukou poradíš, vysvětlíš?

Reply:
S čím přesně? Chybí tam podmět, jak píše Anežka

Reply:
no já jestli jsem to správně napsala, podmět už chápu, ale jestli je správně význam aby to někdo nepoužíval. A jak prosím tě nazvat jakoby ty proměnné, seznam, slovník... jaký je nadpis,  ať napíšu - Pokud chceme, aby někdo ve zbytku kódu nepoužíval konkrétní ...  objekt?

Reply:
Nejobecnější je asi říct "konkrétní *jméno*" (zahrnuje proměnnou, atribut, třídu, ...)